test/build/50_localregistry.bats

```diff
@@ -0,0 +1,90 @@
+load ../common
+
+# shellcheck disable=SC2034
+tag='ch-image push'
+
+# Note: These tests use a local registry listening on localhost:5000 but do
+# not start it. Therefore, they do not depend on whether the pushed images are
+# already present.
+
+
+setup () {
+    scope standard
+    [[ $CH_BUILDER = ch-image ]] || skip 'ch-image only'
+    # Skip unless CI or there is a listener on localhost:5000.
+    if [[ -z $CI ]] && ! (   command -v ss > /dev/null 2>&1 \
+                          && ss -lnt | grep -F :5000); then
+        skip 'no local registry'
+    fi
+
+    # WARNING: If you came here looking for a way to non-interactively
+    # authenticate with ch-image, be aware that these environment variables
+    # are currently undocumented and unsupported.
+    export CH_IMAGE_USERNAME=charlie
+    export CH_IMAGE_PASSWORD=test
+}
+
+@test "${tag}: without destination reference" {
+    # FIXME: This test sets up an alias tag manually so we can use it to push.
+    # Remove when we have real aliasing support for images.
+    ln -vfns 00_tiny "$CH_IMAGE_STORAGE"/img/localhost:5000%00_tiny
+
+    run ch-image -v --tls-no-verify push localhost:5000/00_tiny
+    echo "$output"
+    [[ $status -eq 0 ]]
+    [[ $output = *'pushing image:   localhost:5000/00_tiny'* ]]
+    [[ $output = *"image path:      ${CH_IMAGE_STORAGE}/img/localhost:5000%00_tiny"* ]]
+
+    rm "$CH_IMAGE_STORAGE"/img/localhost:5000%00_tiny
+}
+
+@test "${tag}: with destination reference" {
+    run ch-image -v --tls-no-verify push 00_tiny localhost:5000/00_tiny
```

Comment:
Should we also be testing destination with a paths? What about non-latest tags? For example:

```
00_tiny localhost:5000/tiny_images/00_tiny:tiniest
```

Reply:
Yeah, probably.